### PR TITLE
android: Update target SDK version to 31, i.e., Android 12

### PR DESCRIPTION
**Pull request by chrisbobbe**
Fixes: #5101
Fixes: #5171

### CODE REVIEW

**Comment by chrisbobbe** [comment]
Please also see #5175 for some further improvements to toast texts, and more.

File: src/action-sheets/index.js [hunk 1]
```diff
@@ -587,3 +624,53 @@ export const showStreamActionSheet = ({
     }),
   );
 };
+
+export const showPmActionSheet = ({
```
Comment (gnprice) [inline src/action-sheets/index.js:628]: Really more like "PM *conversation* action sheet", right? For an individual PM, there's the message action sheet.

Comment (gnprice) [reply src/action-sheets/index.js:628]: (Similarly for the function constructing the buttons.)

File: src/compose/ComposeBox.js [hunk 2]
```diff
@@ -259,11 +259,11 @@ class ComposeBoxInner extends PureComponent<Props, State> {
         try {
           response = await api.uploadFile(auth, attachments[i].uri, fileName);
         } catch (e) {
-          showToast(_('Uploading {fileName} failed.', { fileName }));
+          showToast(_('Failed to upload file: {fileName}.', { fileName }));
           this.setMessageInputValue(
             this.state.message.replace(
               placeholder,
-              `[${_('Uploading {fileName} failed.', { fileName })}]()`,
+              `[${_('Failed to upload file: {fileName}.', { fileName })}]()`,
```
Comment (gnprice) [inline src/compose/ComposeBox.js:266]: Let's also drop the `.` at the end. That works when the filename is in the middle of the sentence, but here it risks looking like part of the filename; and in general I think we don't put periods at the end of these short messages.

File: src/chat/PmConversationDetailsScreen.js [hunk 3]
```diff
@@ -15,11 +15,14 @@ import { navigateToAccountDetails } from '../actions';
...
 type Props = $ReadOnly<{|
-  navigation: AppNavigationProp<'group-details'>,
-  route: RouteProp<'group-details', {| recipients: PmKeyRecipients |}>,
+  navigation: AppNavigationProp<'pm-conversation-details'>,
+  route: RouteProp<'pm-conversation-details', {| recipients: PmKeyRecipients |}>,
 |}>;
 
-export default function GroupDetailsScreen(props: Props): Node {
+/**
+ * Details about a PM conversation, identified by PM key recipients.
```
Comment (gnprice) [inline src/chat/PmConversationDetailsScreen.js:23]: The second half of this is redundant with the type. Best to leave it out -- that way it doesn't become wrong if we change the type in the future (renaming it, or changing to something like `PmNarrow`, or something else.)

**Review by gnprice — COMMENTED** [review]
Thanks for taking care of this! Looks great; small comments below.

**Comment by chrisbobbe** [comment]
Thanks! Revision pushed.

**Comment by gnprice** [comment]
Thanks! Looks good -- merging.

**Comment by chrisbobbe** [comment]
Thanks!

**Pull request merged**